superflore/generators/ebuild/ebuild.py

```diff
@@ -263,34 +256,32 @@ def get_ebuild_text(self, distributor, license_text, die_msg=None):
             ret += "    EPATCH_SOURCE=\"${FILESDIR}\""
             ret += " EPATCH_SUFFIX=\"patch\" \\\n"
             ret += "    EPATCH_FORCE=\"yes\" epatch\n"
-            ret += "ros-cmake_src_prepare\n"
+            ret += "    ros-cmake_src_prepare\n"
             ret += "}\n\n"
 
-        # If we're writing the ebuild for catkin, don't build in binary mode.
-        binary_package = '0' if self.name == 'catkin' else '1'
-
         special_pkgs = ['catkin', 'opencv3', 'stage']
-        catkin_cmake_args = "    local mycmakeargs=(\n"\
-            + "        -DCMAKE_INSTALL_PREFIX=${D%/}${ROS_PREFIX}\n"\
-            + "        -DCMAKE_PREFIX_PATH=${ROS_PREFIX}\n"\
-            + "        -DPYTHON_INSTALL_DIR=lib64/python3.5/site-packages\n"\
+        catkin_cmake_args = "local sitedir=\"$(python_get_sitedir)\"\n"\
+            + "    local mycmakeargs=(\n"\
+            + "        -DCMAKE_INSTALL_PREFIX=/${ROS_PREFIX}\n"\
+            + "        -DCMAKE_PREFIX_PATH=/${ROS_PREFIX}\n"\
+            + "        -DPYTHON_INSTALL_DIR=${sitedir#${EPREFIX}}/${ROS_PREFIX}\n"\
             + "        -DCATKIN_BUILD_BINARY_PACKAGE=0\n"\
             + "    )\n"
         # source configuration
         if self.name in special_pkgs:
             ret += "src_configure() {\n"
-            if self.name == 'openvc3':
+            if self.name == 'opencv3':
                 ret += "    filter-flags '-march=*' '-mcpu=*' '-mtune=*'\n"
             elif self.name == 'stage':
                 ret += "    filter-flags '-std=*'\n"
             elif self.name == 'catkin':
                 ret += catkin_cmake_args
-            ret += "    cmake-utils_src_configure\n"
+            ret += "    python_foreach_impl ros-cmake_src_configure_internal\n"
             ret += "}\n\n"
 
         if self.name == 'catkin':
             ret += "src_compile() {\n"
-            ret += "    ${CC} ${FILESDIR}/ros-python.c"
+            ret += "    gcc ${FILESDIR}/ros-python.c"
```

Comment:
@Alessandro-Barbieri For some reason, `${CC}` was not working, so this has been reverted.